src/components/Toolbar.js

```diff
@@ -85,9 +91,15 @@ const Toolbar = ({ cursorOnTableView, cursorOnAlphabeticalSort, cursorPinOpen, c
   const [rightArrowElementClassName = 'hidden', setRightArrowElementClassName] = useState()
   const [overlayName, setOverlayName] = useState()
   const [overlayDescription, setOverlayDescription] = useState()
+  const [toolbarOrigin, updateToolbarOrigin] = useState(toolbarOrigins.topLeft)
 
   const fg = dark ? 'white' : 'black'
   // const bg = dark ? 'black' : 'white'
+  const toolbarRef = useRef()
+  useResizeObserver(toolbarRef, () => {
+    const { scrollWidth, clientWidth } = toolbarRef.current
+    updateToolbarOrigin(scrollWidth - clientWidth > TOOLBAR_WRAPPER_DELTA ? toolbarOrigins.topRight : toolbarOrigins.topLeft)
+  })
 
   useEffect(() => {
     if (toolbarOverlay) {
```

Comment:
This side effect is not really needed. We can simply do this. This will eliminate two state variables.

```javascript
  const shortcut = toolbarOverlay ? shortcutById(toolbarOverlay) : null
  const overlayName = shortcut ? shortcut.name : ''
  const overlayDescription = shortcut ? shortcut.description : ''
```

Reply:
That's right. Though there's more work to be done with this component, I think we should do that refactoring in a separate issue.